conjur-oss/templates/deployment.yaml

```diff
@@ -109,14 +109,15 @@ spec:
         {{- end }}
         imagePullPolicy: {{ .Values.image.pullPolicy }}
 {{- if .Values.account.create }}
-{{- if .Release.IsUpgrade }}
-        # For Helm upgrade, we want to make server startup idempotent, i.e.
-        # tolerant of the chart setting 'account.create=true' being reused.
+        # Try to start up the server with account creation. If that fails,
+        # see if the configured account has already been created, and if so,
+        # restart the server without account creation.
         command: ["bash"]
-        args: ["-c", "conjurctl server --account={{ .Values.account.name }} || conjurctl server"]
-{{ else }}
-        args: ["server", "--account={{ .Values.account.name }}"]
-{{- end }}
+        args: ["-c",
+               "conjurctl server --account={{ .Values.account.name }} \
+                | grep -v 'API key for admin:' \
+                || (conjurctl role retrieve-key {{ .Values.account.name }}:user:admin > /dev/null \
+                    && conjurctl server)"]
```

Comment:
I think this might be easier to read by using the yaml syntax for arrays e.g.
```suggestion
        args:
          - -c
          - |
            conjurctl server --account={{ .Values.account.name }} \
             | grep -v 'API key for admin:' \
             || (conjurctl role retrieve-key {{ .Values.account.name }}:user:admin > /dev/null \
             && conjurctl server)
```

While here.
1. What's the reasoning behind the grep?
2. I think the second arg will result in some weird behaviour. Since 'conjurctl server' creates a long running process, when the first process eventually dies the || might run another server. I think we can avoid that by being more explicit with the branching
   ```
   if conjurctl role retrieve-key {{ .Values.account.name }}:user:admin >/dev/null 2>&1; then
     conjurctl server --account={{ .Values.account.name }}; 
   else 
     conjurctl server;
   fi
   ```

From (2), the final suggestion is

```suggestion
        args:
          - -c
          - |
             if conjurctl role retrieve-key {{ .Values.account.name }}:user:admin >/dev/null 2>&1; then
               conjurctl server --account={{ .Values.account.name }}; 
             else 
               conjurctl server;
             fi
```

Reply:
@doodlesbykumbi , yes, the YAML format is much neater!
And running the `conjurctl role retrieve-key ...` command first makes sense! I was confusing this with earlier attempts to run `conjurctl account create ...`, which fails unconditionally if run before the Conjur server.

The purpose of the `grep -v 'API key for admin'` is to redact the display of the the admin API key in the Conjur log output when the account is auto-created, but to leave all other logs intact. Of course, it's assumed that the Conjur pod will be running in an isolated namespace access limited by Kubernetes RBAC, so maybe this isn't necessary? On the other hand, it would be handy if Conjur server had a command line option to skip displaying this info.
WDYT?